src/hit/api/params.cpp

```diff
@@ -42,22 +42,34 @@ namespace hit {
         return x - repr(x, n) + 1;
     }
 
-    uint64_t nextPrime(uint64_t last_prime, uint64_t two_n, const vector<Modulus> &mods) {
-        // get the closest prime to `last_prime` that is also congruent to 1 mod 2N
-        uint64_t next_test_val = to1Coset(last_prime);
+    // Find the smallest prime larger than `x` that is not in `mods`.
+    uint64_t nextPrime(uint64_t x, uint64_t n, const vector<Modulus> &mods) {
+        // find a number near x that is congruent to 1 mod n
+        uint64_t next_test_val = to1Coset(x, n);
+
+        // to1Coset can return a value smaller than x
+        if (next_test_val < x) {
+            next_test_val += n;
+        }
 
         while (!miller_rabin_test(next_test_val, 25) || contains(next_test_val, mods)) {
-            next_test_val += two_n;
+            next_test_val += n;
         }
         return next_test_val;
     }
 
-    uint64_t prevPrime(uint64_t last_prime, uint64_t two_n, const vector<Modulus> &mods) {
-        // get the closest prime to `last_prime` that is also congruent to 1 mod 2N
-        uint64_t next_test_val = to1Coset(last_prime);
+    // Find the largest prime smaller than `x` that is not in `mods`.
+    uint64_t prevPrime(uint64_t x, uint64_t n, const vector<Modulus> &mods) {
+        // find a number near x that is congruent to 1 mod n
+        uint64_t next_test_val = to1Coset(x, n);
+
+        // to1Coset can return a value larger than x
+        if (next_test_val > x) {
+            next_test_val -= n;
+        }
 
         while (!miller_rabin_test(next_test_val, 25) || contains(next_test_val, mods)) {
```

Comment:
I assume that `contains` is much faster than `miller_rabin_test`, so should we put it before the `||` so we short-circuit evaluation and skip the miller-rabin test if not needed?